### PR TITLE
Add materialized view for logged in/out post views

### PR DESCRIPTION
**Pull request by Justin-Holmes**

### CODE REVIEW

**Comment by ellodanger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a linked tracker story for this PR in the description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: lib/tasks/views.rake [hunk 1]
```diff
@@ -4,5 +4,6 @@ namespace :views do
     Scenic.database.refresh_materialized_view('impressions_by_stream_by_day', concurrently: true)
     Scenic.database.refresh_materialized_view('impressions_by_category_by_days', concurrently: true)
     Scenic.database.refresh_materialized_view('impressions_by_email_by_days', concurrently: true)
+    Scenic.database.refresh_materialized_view('impressions_by_logged_in_and_out_by_days', concurrently: true)
```
Comment (jayzes) [inline lib/tasks/views.rake]: Do we need a migration and a unique index to get this into place?

Comment (Justin-Holmes) [inline lib/tasks/views.rake]: @jayzes Ah, yes we do. I'm assuming we can't add a unique index on the logged out/in aggregate functions.

**Comment by ellodanger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a linked tracker story for this PR in the description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by Justin-Holmes** [comment]
@jayzes Added the migration. Do you think we should increase the timeout? Fingers crossed this one doesn't give us any trouble.

File: lib/tasks/views.rake (hunk 1 above)
Comment (jayzes) [inline lib/tasks/views.rake]: Yup, I think leaving the index by day should be totally fine for this one. Just needs to be unique per row, and our rows are date-based.

**Comment by jayzes** [comment]
@Justin-Holmes let's give it a try and see what it does. if it fails, we could explore using `WITH NO DATA` (https://www.postgresql.org/docs/9.3/static/sql-creatematerializedview.html) though I'm not 100% sure how Scenic interacts with that.

**Pull request merged**